packages/core/src/forms/form-group/form-group.element.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { LitElement, html } from 'lit-element';
+import { baseStyles, querySlotAll, syncDefinedProps, property } from '@clr/core/internal';
+import { CdsInternalControlGroup } from '../control-group/control-group.element.js';
+import { CdsInternalControlInline } from '../control-inline/control-inline.element.js';
+import { CdsControl } from '../control/control.element.js';
+import { FormLayout, ControlWidth } from '../utils/interfaces.js';
+import { getLargestPrimaryLabelWidth } from '../utils/index.js';
+import { styles } from './form-group.element.css.js';
+
+/**
+ * Form
+ *
+ * ```typescript
+ * import '@clr/core/forms/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-form-group layout="horizontal">
+ *   <cds-input>
+ *     <label>Test</label>
+ *     <input type="text" />
+ *   </cds-input>
+ *
+ *   <cds-input>
+ *     <label>Test</label>
+ *     <input type="text" />
+ *   </cds-input>
+ * </cds-form-group>
+ * ```
+ *
+ * @element cds-form
+ * @slot default - For projecting input controls
+ */
+export class CdsFormGroup extends LitElement {
+  /**
+   * @type {horizontal | horizontal-inline | vertical | vertical-inline | compact}
```

Comment:
The `@type` annotations are for the web component analyzer tool that generates our API. The tool does not expand on `types` so without the annotation the API docs would only show `FormLayout` instead of each option available. This can go away one the tool can update to support this. We only need the annotations on types and interfaces. Stand alone types work as expected (string, boolean, number, ect...)